rtl/eclrtl/eclrtl.cpp

```diff
@@ -4892,6 +4894,50 @@ class CStrRegExprFindInstance : implements IStrRegExprFindInstance
         }
     }
 
+    bool nextMatch()
+    {
+        if (!matched)
+            return false;
+
+        const char *start = subs[0].second==subs[0].first?subs[0].second+1: subs[0].second;
+        if (start >= sample_end)
+            return (matched = false);
+
+        matched = boost::regex_search(start, subs, *regEx);
+
+        return matched;
+    }
+
+    void getMatchSet(bool  & __isAllResult, size32_t & __resultBytes, void * & __result)
+    {
+        rtlRowBuilder out;
+        byte *outData;
+        size32_t outBytes = 0;
+
+        boost::regex_iterator<char *> cur(sample, sample_end, *regEx);
+        boost::regex_iterator<char *> end;
+
+        for (; cur != end; ++cur)
```

Comment:
The default constructor for regex_iterator is an end of sequence marker:  http://www.boost.org/doc/libs/1_55_0/libs/regex/doc/html/boost_regex/ref/regex_iterator.html